w3af/plugins/grep/expect_ct.py

```diff
@@ -0,0 +1,90 @@
+"""
+expect_ct.py
+
+Copyright 2019 Andres Riancho
+
+This file is part of w3af, http://w3af.org/ .
+
+w3af is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation version 2 of the License.
+
+w3af is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with w3af; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+"""
+from w3af.core.data.kb.info import Info
+from w3af.core.data.kb.info_set import InfoSet
+from w3af.core.controllers.plugins.grep_plugin import GrepPlugin
+
+ECT_HEADER = 'Expect-CT'
+
+class expect_ct(GrepPlugin):
+    """
+    Check if HTTPS responses have the Expect-CT header set.
+
+    """
+    def __init__(self):
+        super(expect_ct, self).__init__()
+        self._reports = 0
+
+    def grep(self, request, response):
+        """
+        Check if HTTPS responses have the Expect-CT header set.
+
+        :param request: The HTTP request object.
+        :param response: The HTTP response object
+        :return: None, all results are saved in the kb.
+        """
+
+        if request.get_url().get_protocol() != 'https':
+            return
+
+        expect_ct_header_value, _ = response.get_headers().iget(ECT_HEADER, None)
+        if expect_ct_header_value is not None:
+            return
+
+        self._reports += 1
```

Comment:
You know what? I made a mistake with the previous comment on `self._reports`. I was thinking only about functionality: InfoSet will take care of the max reports, so `self._reports` is not required on that side.... **but if we think about performance** then the story is different! By using `self._reports` you're preventing a lot of processing after the 50th reported vulnerability, which is great!

Please re-add the self._reports check at the beginning of this method.

Sorry for this